aya/src/programs/links.rs

```diff
@@ -394,6 +397,125 @@ pub enum LinkError {
     SyscallError(#[from] SyscallError),
 }
 
+#[derive(Debug)]
+pub(crate) enum LinkRef {
+    Id(u32),
+    Fd(RawFd),
+}
+
+bitflags::bitflags! {
+    /// Flags which are use to build a set of MprogOptions.
+    #[derive(Clone, Copy, Debug, Default)]
+    pub(crate) struct MprogFlags: u32 {
+        const REPLACE = BPF_F_REPLACE;
+        const BEFORE = BPF_F_BEFORE;
+        const AFTER = BPF_F_AFTER;
+        const ID = BPF_F_ID;
+        const LINK = BPF_F_LINK;
+    }
+}
+
+/// Arguments required for interacting with the kernel's multi-prog API.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 6.6.0.
+///
+/// # Example
+///
+///```no_run
+/// # let mut bpf = aya::Ebpf::load(&[])?;
```

Comment:
```suggestion
/// # let mut ebpf = aya::Ebpf::load(&[])?;
```

Reply:
We're still `let mut bpf` in all other examples. I'd prefer to stick to the established pattern.. or change all of them to be consistent.

Reply:
Fair, let's change them in a separate PR then.